src/FFmpegWriter.cpp

```diff
@@ -691,15 +697,8 @@ void FFmpegWriter::WriteFrame(std::shared_ptr<Frame> frame) {
 
 	// Write the frames once it reaches the correct cache size
 	if ((int)spooled_video_frames.size() == cache_size || (int)spooled_audio_frames.size() == cache_size) {
-		// Is writer currently writing?
-		if (!is_writing)
-			// Write frames to video file
-			write_queued_frames();
-
-		else {
-			// Write frames to video file
-			write_queued_frames();
-		}
+		// Write frames to video file
+		write_queued_frames();
 	}
```

Comment:
This was the bit Codecov flagged as unreachable. I should certainly say so! Turns out this block has a bit of history.

Once upon a time it looked basically exactly as it will after my change. Then, 8 years ago in 0f3758bc29d2470d0cf2fe589e89b21330aee0d9, it became this:
```c++
	if (spooled_video_frames.size() == cache_size || spooled_audio_frames.size() == cache_size)
	{
		// Is writer currently writing?
		if (!is_writing)
			// Write frames to video file
			write_queued_frames();

		else
		{
			// YES, WRITING... so wait until it finishes, before writing again
			while (is_writing)
				usleep(250 * 1000); // sleep for 250 milliseconds

			// Write frames to video file
			write_queued_frames();
		}

	}
```

It trundled along like that with only minor changes until 851ad86634c01a2c2c05bdffc3805466b3354c91, when (despite the commit message being "Changing some sleep() calls to usleep()", the sleep was _removed_ entirely, making the two branches accidentally equivalent.

Maybe the right change is to put the sleep call _back_, I don't know. But in its current state the code is nonsensical, the second branch can never be reached.

Reply:
(I might also point out that even _if_ we want to put the sleep-before-write back in, the cleanest form would be...)
```c++
if (spooled_video_frames.size() == cache_size || spooled_audio_frames.size() == cache_size) {
    // Wait until an in-progress write finishes
    while (is_writing) {
        std::this_thread::sleep_for(std::chrono::milliseconds(2500));
    }
    write_queued_frames();
}
```
There's no reason for the preceding `if()`, when `is_writing` is false execution will fall straight through the `while()` without a single iteration.